src/cript/nodes/exceptions.py

```diff
@@ -57,3 +59,37 @@ def __init__(self, node_type: str, json_dict: str):
 
     def __str__(self):
         return f"JSON Serialization failed for node type {self.node_type} with JSON dict: {self.json_str}"
+
+
+class CRIPTOrphranedNodesError(CRIPTException, ABC):
+    def __init__(self, orphraned_node: "BaseNode"):
+        self.orphaned_node = orphraned_node
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+
+class CRIPTOrphranedMaterialError(CRIPTOrphranedNodesError):
+    def __init__(self, orphaned_node: "Material"):
+        super().__init__(orphaned_node)
+
+    def __str__(self):
+        ret_string = "While validating a project graph, an orphraned material node was found. "
+        ret_string += "This material is present in the graph, but not listed in the project. "
+        ret_string += "Please add the node like: `your_project.materials += [orphaned_material]`. "
+        ret_string += f"The orphraned material was {self.orphraned_node}."
+        return ret_string
+
+
+class CRIPTOrphranedExperimentError(CRIPTOrphranedNodesError):
```

Comment:
can we make one error `OrphanedNodeError` and just change the messages depending on how it is called?

I think that could make lives easier and we don't have to keep adding more exceptions for every new orphaned node, we just have one and we make it into a template and depending on how its called it gives the correct context for it

Reply:
No, on the contrary.
I actually want to diversify this error more.

We still have a common base node (and I will put most of the logic there), but different exceptions help a lot in fixing the error.

Consider this.

```python
active_experiment = project.find_children({"node": ["Experiment"]})[0]
try:
   project.validate()
except CRIPTOrphanedMaterialError as exc:
  project.materials += exc.orphaned_node
except CRIPTOrphanedDataError as exc:
  active_experiment.data += exc.orphaned_node
except CRIPTOrphanedProcessError as exc:
  active_experiment.process += exc.orphaned_node
  ```
  
  The second and thrid exception handling block is only possible if you can easily deduce the orphaned node type from the exception.

Reply:
see at the helper function, that I added. It actively uses this.